src/main/scala/cromwell/engine/backend/ExecutionResult.scala

```diff
@@ -21,8 +21,13 @@ final case class SuccessfulBackendCallExecution(outputs: CallOutputs, executionE
  */
 case object AbortedExecution extends ExecutionResult
 
-/**
- * Failed execution, possibly having a return code.
- */
-final case class FailedExecution(e: Throwable, returnCode: Option[Int] = None) extends ExecutionResult
+sealed trait FailedExecution extends ExecutionResult {
+  def e: Throwable
```

Comment:
Currently I don't think the type is actually checked anywhere.
However we can imagine that there would be different behaviors in the WA depending on the type of failure.
For retryable errors for instance we might want to do something differently if it's been `Preempted` or `WhateverOtherResasonTriggeringARetry` ?